File: 4178-vowel-consonant-score/vowel-consonant-score.cpp
class Solution {
public:
    int vowelConsonantScore(string s) {
        s.erase(remove(s.begin(),s.end(),' '),s.end());
        int v=0;
        int c=0;
        for(char ch:s){
            if(!isdigit(ch) && isVowel(ch)){
                v++;
            }
            else if(!isdigit(ch) && !isVowel(ch)){
                c++;
            }
        }
        if(c>0){
            return floor((double)v/c);
        }
        else{
            return 0;
        }
    }
    bool isVowel(char ch){
        return ch=='a'||ch=='e'||ch=='i'||ch=='o'||ch=='u';
    }
};

File: 4178-vowel-consonant-score/vowel-consonant-score.java
class Solution {
    public int vowelConsonantScore(String s) {
        s=s.replace(" ","");
        int v=0;
        int c=0;
        for(char ch:s.toCharArray()){
            if(!Character.isDigit(ch) && isVowel(ch)){
                v++;
            }
            else if(!Character.isDigit(ch) && !isVowel(ch)){
                c++;
            }
        }
        if(c>0){
            return Math.floorDiv(v,c);
        }
        else{
            return 0;
        }
    }
    public boolean isVowel(char ch){
        return ch=='a'||ch=='e'||ch=='i'||ch=='o'||ch=='u';
    }
}
